### ISSUE TITLE
JetEngine - Layout Switcher Beta

### ISSUE THREAD

**Issue by MjHead** [CrocoBlock/suggestions#7100]
Please note the Layout Switcher widget is now in Beta, so it is installed independently from the JetEngine core. In future releases, the widget will become a part of the JetEngine plugin, and you will be able to deactivate/activate it in the External Modules section. At the moment you can download it from you Crocoblock account, Beta Releases section - https://tppr.me/JAfyj

### Overview

**Layout Switcher** is a widget allowing you to override some **Listing Grid** settings. We placed this feature into a separate widget for design flexibility purposes.

**Layout Switcher** represents a list of layouts and their settings, which can be overridden in the **Listing Grid**. Currently, you can override the set Listing Template and Columns Number.

The switcher is attached to the Listing Grid widget by the widget's unique ID. This feature is very popular on eCommerce sites, where a switcher is often added to change the Product List layout, for instance, Grid/List.

### Logic

When you change the layout, the settings specified in the Layout Switcher as overridden will change in the Listing Grid. If the layout overrides the Listing Template, the listing will be changed via ajax. If it overrides just the number of columns, the change will happen immediately.

When changing the layout, the current layout is saved in cookies, and when you reload the page, the last selected layout remains applied to the  **Listing Grid**.

### Widget settings

https://tppr.me/7rzVF

1. **Select a Listing Grid widget** – in this option, choose a Listing Grid widget from the ones located on the page to which you want to attach the Layout Switcher. The list of widget names in this control is formed according to the following template – %widget_name% (%widget_uniq_id%) - %widget_custom_title% - %widget_css_id% https://tppr.me/BZgzR, %widget_custom_title% – can be set in Navigator https://tppr.me/9Q5GV, %widget_css_id% – in the Advanced tab of the widget https://tppr.me/WGCEx. We recommend adding a custom title or CSS ID to the widget – in that case, you’ll be able to find it in the list faster. 
2. **Layouts** – contains the list of layouts of the switcher: 
  - Content tab:
    - Label – a title of the layout which will be displayed in the Layout button;
    - Slug – a slug of the layout. The active layout will be stored in cookies by this slug, so it must be unique. If empty, the slug will be generated automatically from the Label;
    - Icon – an icon of the layout.
  -  Settings tab:
    - Is Default Layout – this option allows us to set the default Layout and its settings, which are the default settings of the Listing Grid widget to which you attached the Switcher. One of the layouts must be set as default;
    - Listing – set here a listing template that will override the default listing settings;
    - Columns Number – indicate the desired number of columns within the selected layout style.
3. **Labels** – Show/Hide – this option hides the labels of the Switcher buttons. This option is needed to display only icons in the buttons, and the labels will be added to the aria-label attribute of the buttons for accessibility purposes.

### Feedback
Please add your feedback in the comments below. It will be very important for use before merge it into the JetEngine

**Comment by perks98** [comment]
I have just tried it for a potential project and noticed the column number input doesn't work when JetEngine listing is set to 'Auto'.

**Comment by kaskad88** [comment]
> I have just tried it for a potential project and noticed the column number input doesn't work when JetEngine listing is set to 'Auto'.

@perks98 Thanks for your report! We will fix it.

**Comment by Lonsdale201** [comment]
Hi. It's not necessarily a bug, but after setting the secondary, so not default listing in the layout system, it would be useful if "Edit With jetplugins" would detect it. :)

**Comment by ludovicallo** [comment]
Hi. Is it Bricks friendly ?

**Comment by kaskad88** [comment]
> Hi. Is it Bricks friendly ?

We plan to do it

**Comment by Paul-Heron** [comment]
If you switch layouts it breaks the listing grid infinite scroll until you reload the page.

**Comment by Maxican** [comment]
I would like to give it a try, but it disappeared from the account page. I can only see my purchased plugins and some themes.

<img width="608" alt="image" src="https://github.com/Crocoblock/suggestions/assets/3306841/b9748cc7-51bc-458b-aa36-4ab2da4c0619">

**Comment by Paul-Heron** [comment]
> I would like to give it a try, but it disappeared from the account page. I can only see my purchased plugins and some themes.

If you go to the code tab at the top of the page and then click the green code button > download as zip.
